### PR TITLE
Terminology style updates for auth book

### PR DESCRIPTION
**Pull request by bergerhoffer**
Preview (internal): http://file.rdu.redhat.com/~ahoffer/2020/term-updates-auth/authentication/understanding-authentication.html

### CODE REVIEW

File: modules/identity-provider-basic-authentication-CR.adoc [hunk 1]
```diff
@@ -32,12 +32,12 @@ spec:
 ----
 <1> This provider name is prefixed to the returned user ID to form an identity
 name.
-<2> Controls how mappings are established between this provider's identities and user objects.
+<2> Controls how mappings are established between this provider's identities and `User` objects.
 <3> URL accepting credentials in Basic authentication headers.
-<4> Optional: Reference to an {product-title} ConfigMap containing the
+<4> Optional: Reference to an {product-title} `ConfigMap` containing the
```
Comment (jboxman) [inline modules/identity-provider-basic-authentication-CR.adoc:37]: `ConfigMap` object

File: modules/identity-provider-basic-authentication-CR.adoc [hunk 2]
```diff
@@ -32,12 +32,12 @@ spec:
...
 PEM-encoded certificate authority bundle to use in validating server
 certificates for the configured URL.
-<5> Optional: Reference to an {product-title} Secret containing the client
+<5> Optional: Reference to an {product-title} `Secret` containing the client
```
Comment (jboxman) [inline modules/identity-provider-basic-authentication-CR.adoc:40]: `Secret` object

File: modules/identity-provider-basic-authentication-CR.adoc [hunk 3]
```diff
@@ -32,12 +32,12 @@ spec:
...
 certificate to present when making requests to the configured URL.
-<6> Reference to an {product-title} Secret containing the key for the
+<6> Reference to an {product-title} `Secret` containing the key for the
```
Comment (jboxman) [inline modules/identity-provider-basic-authentication-CR.adoc:42]: `Secret` object

File: modules/identity-provider-config-map.adoc [hunk 4]
```diff
@@ -8,15 +8,15 @@
...
-= Creating a ConfigMap
+= Creating a config map
 
-Identity providers use {product-title} ConfigMaps in the `openshift-config`
+Identity providers use {product-title} `ConfigMap` objects in the `openshift-config`
 namespace to contain the certificate authority bundle. These are primarily
 used to contain certificate bundles needed by the identity provider.
 
-* Define an {product-title} ConfigMap containing the
+* Define an {product-title} `ConfigMap` containing the
```
Comment (jboxman) [inline modules/identity-provider-config-map.adoc:17]: object

File: modules/identity-provider-config-map.adoc [hunk 5]
```diff
@@ -8,15 +8,15 @@
...
 certificate authority by using the following command. The certificate
-authority must be stored in the `ca.crt` key of the ConfigMap.
+authority must be stored in the `ca.crt` key of the `ConfigMap`.
```
Comment (jboxman) [inline modules/identity-provider-config-map.adoc:19]: object

File: modules/identity-provider-configuring-apache-request-header.adoc [hunk 6]
```diff
@@ -21,20 +21,20 @@ You must have the following packages installed on your local machine:
 ** `mod_auth_gssapi`
 
 * Generate a CA for validating requests that submit the trusted header. Define
-an {product-title} ConfigMap containing the CA. This is done by running:
+an {product-title} `ConfigMap` containing the CA. This is done by running:
```
Comment (jboxman) [inline modules/identity-provider-configuring-apache-request-header.adoc:24]: object

File: modules/identity-provider-configuring-apache-request-header.adoc [hunk 7]
```diff
@@ -21,20 +21,20 @@ You must have the following packages installed on your local machine:
...
 +
 [source,terminal]
 ----
 $ oc create configmap ca-config-map --from-file=ca.crt=/path/to/ca -n openshift-config
 ----
 +
-The CA must be stored in the `ca.crt` key of the ConfigMap.
+The CA must be stored in the `ca.crt` key of the `ConfigMap`.
```
Comment (jboxman) [inline modules/identity-provider-configuring-apache-request-header.adoc:31]: object

File: modules/identity-provider-github-CR.adoc [hunk 8]
```diff
@@ -37,20 +37,20 @@ spec:
 ----
 <1> This provider name is prefixed to the GitHub numeric user ID to form an
 identity name. It is also used to build the callback URL.
-<2> Controls how mappings are established between this provider's identities and user objects.
-<3> Optional: Reference to an {product-title} ConfigMap containing the
+<2> Controls how mappings are established between this provider's identities and `User` objects.
+<3> Optional: Reference to an {product-title} `ConfigMap` containing the
```
Comment (jboxman) [inline modules/identity-provider-github-CR.adoc:41]: object

File: modules/identity-provider-github-CR.adoc [hunk 9]
```diff
@@ -37,20 +37,20 @@ spec:
...
 PEM-encoded certificate authority bundle to use in validating server
 certificates for the configured URL. Only for use in GitHub Enterprise
 with a non-publicly trusted root certificate.
 <4> The client ID of a
 link:https://github.com/settings/applications/new[registered GitHub OAuth
 application]. The application must be configured with a callback URL of
 `\https://oauth-openshift.apps.<cluster-name>.<cluster-domain>/oauth2callback/<idp-provider-name>`.
-<5> Reference to an {product-title} Secret containing the client secret
+<5> Reference to an {product-title} `Secret` containing the client secret
```
Comment (jboxman) [inline modules/identity-provider-github-CR.adoc:49]: object

File: modules/identity-provider-gitlab-CR.adoc [hunk 10]
```diff
@@ -31,15 +31,15 @@ spec:
...
 <1> This provider name is prefixed to the GitLab numeric user ID to form an
 identity name. It is also used to build the callback URL.
-<2> Controls how mappings are established between this provider's identities and user objects.
+<2> Controls how mappings are established between this provider's identities and `User` objects.
 <3> The client ID of a
 link:https://docs.gitlab.com/ce/api/oauth2.html[registered GitLab OAuth application].
 The application must be configured with a callback URL of
 `\https://oauth-openshift.apps.<cluster-name>.<cluster-domain>/oauth2callback/<idp-provider-name>`.
-<4> Reference to an {product-title} Secret containing the client secret
+<4> Reference to an {product-title} `Secret` containing the client secret
```
Comment (jboxman) [inline modules/identity-provider-gitlab-CR.adoc:39]: object

File: modules/identity-provider-gitlab-CR.adoc [hunk 11]
```diff
@@ -31,15 +31,15 @@ spec:
...
 issued by GitLab.
 <5> The host URL of a GitLab provider. This could either be `\https://gitlab.com/`
 or any other self hosted instance of GitLab.
-<6> Optional: Reference to an {product-title} ConfigMap containing the 
+<6> Optional: Reference to an {product-title} `ConfigMap` containing the
```
Comment (jboxman) [inline modules/identity-provider-gitlab-CR.adoc:43]: object

File: modules/identity-provider-google-CR.adoc [hunk 12]
```diff
@@ -29,11 +29,11 @@ spec:
 ----
 <1> This provider name is prefixed to the Google numeric user ID to form an
 identity name. It is also used to build the redirect URL.
-<2> Controls how mappings are established between this provider's identities and user objects.
+<2> Controls how mappings are established between this provider's identities and `User` objects.
 <3> The client ID of a link:https://console.developers.google.com/[registered
 Google project]. The project must be configured with a redirect URI of
 `\https://oauth-openshift.apps.<cluster-name>.<cluster-domain>/oauth2callback/<idp-provider-name>`.
-<4> Reference to an {product-title} Secret containing the client secret
+<4> Reference to an {product-title} `Secret` containing the client secret
```
Comment (jboxman) [inline modules/identity-provider-google-CR.adoc:36]: object

File: modules/identity-provider-htpasswd-secret.adoc [hunk 13]
```diff
@@ -14,7 +14,7 @@ contains the HTPasswd user file.
 
 .Procedure
 
-* Create an {product-title} Secret that contains the HTPasswd users file.
+* Create an {product-title} `Secret` that contains the HTPasswd users file.
```
Comment (jboxman) [inline modules/identity-provider-htpasswd-secret.adoc:17]: object

File: modules/identity-provider-htpasswd-update-users.adoc [hunk 14]
```diff
@@ -9,14 +9,14 @@ You can add or remove users from an existing HTPasswd identity provider.
 
 .Prerequisites
 
-* You have created a secret that contains the HTPasswd user file. This procedure assumes that it is named `htpass-secret`.
+* You have created a `Secret` that contains the HTPasswd user file. This procedure assumes that it is named `htpass-secret`.
```
Comment (jboxman) [inline modules/identity-provider-htpasswd-update-users.adoc:12]: object

File: modules/identity-provider-htpasswd-update-users.adoc [hunk 15]
```diff
@@ -9,14 +9,14 @@ You can add or remove users from an existing HTPasswd identity provider.
...
 * You have configured an HTPasswd identity provider. This procedure assumes that it is named `my_htpasswd_provider`.
 * You have access to the `htpasswd` utility. On Red Hat Enterprise Linux this is available by installing the `httpd-tools` package.
 * You have cluster administrator privileges.
 
 .Procedure
 
-. Retrieve the HTPasswd file from the `htpass-secret` secret and save the file to your file system:
+. Retrieve the HTPasswd file from the `htpass-secret` `Secret` and save the file to your file system:
```
Comment (jboxman) [inline modules/identity-provider-htpasswd-update-users.adoc:19]: object

File: modules/identity-provider-htpasswd-update-users.adoc [hunk 16]
```diff
@@ -60,7 +60,7 @@ $ oc create secret generic htpass-secret --from-file=htpasswd=users.htpasswd --d
 
 . If you removed one or more users, you must additionally remove existing resources for each user.
 
-.. Delete the user:
+.. Delete the `User`:
```
Comment (jboxman) [inline modules/identity-provider-htpasswd-update-users.adoc:63]: object

File: modules/identity-provider-htpasswd-update-users.adoc [hunk 17]
```diff
@@ -51,7 +51,7 @@ $ htpasswd -D users.htpasswd <username>
 Deleting password for user <username>
 ----
 
-. Replace the `htpass-secret` secret with the updated users in the `users.htpasswd` file:
+. Replace the `htpass-secret` `Secret` with the updated users in the `users.htpasswd` file:
```
Comment (jboxman) [inline modules/identity-provider-htpasswd-update-users.adoc:54]: object

File: modules/identity-provider-htpasswd-update-users.adoc [hunk 18]
```diff
@@ -75,7 +75,7 @@ user.user.openshift.io "<username>" deleted
 +
 Be sure to remove the user, otherwise the user can continue using their token as long as it has not expired.
 
-.. Delete the identity for the user:
+.. Delete the `Identity` for the user:
```
Comment (jboxman) [inline modules/identity-provider-htpasswd-update-users.adoc:78]: object

File: modules/identity-provider-keystone-CR.adoc [hunk 19]
```diff
@@ -32,14 +32,14 @@ spec:
         name: client-key-secret
 ----
 <1> This provider name is prefixed to provider user names to form an identity name.
-<2> Controls how mappings are established between this provider's identities and user objects.
+<2> Controls how mappings are established between this provider's identities and `User` objects.
 <3> Keystone domain name. In Keystone, usernames are domain-specific. Only a single domain is supported.
 <4> The URL to use to connect to the Keystone server (required). This must
 use https.
-<5> Optional: Reference to an {product-title} ConfigMap containing the
+<5> Optional: Reference to an {product-title} `ConfigMap` containing the
```
Comment (jboxman) [inline modules/identity-provider-keystone-CR.adoc:39]: object

File: modules/identity-provider-keystone-CR.adoc [hunk 20]
```diff
@@ -32,14 +32,14 @@ spec:
...
 PEM-encoded certificate authority bundle to use in validating server
 certificates for the configured URL.
-<6> Optional: Reference to an {product-title} Secret containing the client
+<6> Optional: Reference to an {product-title} `Secret` containing the client
```
Comment (jboxman) [inline modules/identity-provider-keystone-CR.adoc:42]: object

File: modules/identity-provider-keystone-CR.adoc [hunk 21]
```diff
@@ -32,14 +32,14 @@ spec:
...
 certificate to present when making requests to the configured URL.
-<7> Reference to an {product-title} Secret containing the key for the
+<7> Reference to an {product-title} `Secret` containing the key for the
```
Comment (jboxman) [inline modules/identity-provider-keystone-CR.adoc:44]: object

File: modules/security-context-constraints-rbac.adoc [hunk 22]
```diff
@@ -54,13 +54,13 @@ the `resourceNames` field.
...
 A local or cluster role with such a rule allows the subjects that are
-bound to it with a RoleBinding or a ClusterRoleBinding to use the
+bound to it with a role binding or a cluster role binding to use the
 user-defined SCC called `scc-name`.
 
 [NOTE]
 ====
 Because RBAC is designed to prevent escalation, even project administrators
-will be unable to grant access to an SCC. By defualt, they are not
+are unable to grant access to an SCC. By default, they are not
```
Comment (jboxman) [inline modules/security-context-constraints-rbac.adoc:63]: Does it make sense to elaborate on _they_ here? Is it RBACs or administrators?

File: modules/oauth-configuring-token-inactivity-timeout.adoc [hunk 23]
```diff
@@ -19,9 +19,9 @@ If the token inactivity timeout is also configured in your OAuth client, that va
 
 .Procedure
 
-. Update the OAuth configuration to set a token inactivity timeout.
+. Update the `OAuth` configuration to set a token inactivity timeout.
```
Comment (jboxman) [inline modules/oauth-configuring-token-inactivity-timeout.adoc:22]: Looks like s/configuration/object/ is acceptable here.

File: modules/oauth-configuring-token-inactivity-timeout-clients.adoc [hunk 24]
```diff
@@ -19,9 +19,9 @@ If the token inactivity timeout is also configured in the internal OAuth server
 
 .Procedure
 
-. Update the OAuth client configuration to set a token inactivity timeout.
+* Update the `OAuthClient` configuration to set a token inactivity timeout.
```
Comment (jboxman) [inline modules/oauth-configuring-token-inactivity-timeout-clients.adoc:22]: Looks like s/configuration/object/ is acceptable here.

File: modules/identity-provider-secret.adoc [hunk 25]
```diff
@@ -9,20 +9,20 @@
...
+* You can define an {product-title} `Secret` containing a string
 by using the following command.
 +
 [source,terminal]
 ----
 $ oc create secret generic <secret_name> --from-literal=clientSecret=<secret> -n openshift-config
 ----
 
-* You can define an {product-title} Secret containing the contents of a
+* You can define an {product-title} `Secret` containing the contents of a
```
Comment (jboxman) [inline modules/identity-provider-secret.adoc:25]: object

File: modules/identity-provider-secret.adoc [hunk 26]
```diff
@@ -9,20 +9,20 @@
...
 [id="identity-provider-creating-secret_{context}"]
-= Creating the Secret
+= Creating the secret
 
-Identity providers use {product-title} Secrets in the `openshift-config`
+Identity providers use {product-title} `Secret` objects in the `openshift-config`
 namespace to contain the client secret, client certificates, and keys.
 
-* You can define an {product-title} Secret containing a string
+* You can define an {product-title} `Secret` containing a string
```
Comment (jboxman) [inline modules/identity-provider-secret.adoc:17]: object

File: modules/identity-provider-request-header-CR.adoc [hunk 27]
```diff
@@ -55,7 +55,7 @@ then proxy them to `https://_<namespace_route>_/oauth/authorize`.
 `${url}` is replaced with the current URL, escaped to be safe in a query parameter.
 `${query}` is replaced with the current query string.
 If this attribute is not defined, then `challengeURL` must be used.
-<5> Reference to an {product-title} ConfigMap containing a PEM-encoded
+<5> Reference to an {product-title} `ConfigMap` containing a PEM-encoded
```
Comment (jboxman) [inline modules/identity-provider-request-header-CR.adoc:58]: object

File: modules/identity-provider-oidc-CR.adoc [hunk 28]
```diff
@@ -38,11 +38,11 @@ spec:
 ----
 <1> This provider name is prefixed to the value of the identity claim to form an
 identity name. It is also used to build the redirect URL.
-<2> Controls how mappings are established between this provider's identities and user objects.
+<2> Controls how mappings are established between this provider's identities and `User` objects.
 <3> The client ID of a client registered with the OpenID provider. The
 client must be allowed to redirect to
 `\https://oauth-openshift.apps.<cluster_name>.<cluster_domain>/oauth2callback/<idp_provider_name>`.
-<4> Reference to an {product-title} Secret containing the client secret.
+<4> Reference to an {product-title} `Secret` containing the client secret.
```
Comment (jboxman) [inline modules/identity-provider-oidc-CR.adoc:45]: object

File: modules/identity-provider-ldap-secret.adoc [hunk 29]
```diff
@@ -3,12 +3,12 @@
 // * authentication/identity_providers/configuring-ldap-identity-provider.adoc
 
 [id="identity-provider-creating-ldap-secret_{context}"]
-= Creating the LDAP Secret
+= Creating the LDAP secret
 
-To use the identity provider, you must define an {product-title} Secret
-that contains the bindPassword.
+To use the identity provider, you must define an {product-title} `Secret`
```
Comment (jboxman) [inline modules/identity-provider-ldap-secret.adoc:8]: `Secret` object

File: modules/identity-provider-ldap-secret.adoc [hunk 30]
```diff
@@ -3,12 +3,12 @@
...
+that contains the `bindPassword`.
```
Comment (jboxman) [inline modules/identity-provider-ldap-secret.adoc:9]: field?

File: modules/identity-provider-ldap-secret.adoc [hunk 31]
```diff
@@ -3,12 +3,12 @@
...
 
-* Define an {product-title} Secret that contains the bindPassword.
+* Define an {product-title} `Secret` that contains the `bindPassword`.
```
Comment (jboxman) [inline modules/identity-provider-ldap-secret.adoc:11]: `Secret` object
`bindPassword` field

File: modules/identity-provider-ldap-CR.adoc [hunk 32]
```diff
@@ -69,5 +69,5 @@ URLs always attempt to connect using TLS.
 ====
 To whitelist users for an LDAP integration, use the `lookup` mapping method.
 Before a login from LDAP would be allowed, a cluster administrator must create
-an identity and user object for each LDAP user.
+an `Identity` and `User` object for each LDAP user.
```
Comment (jboxman) [inline modules/identity-provider-ldap-CR.adoc:72]: `Identity` object and `User` object?

File: modules/identity-provider-ldap-CR.adoc [hunk 33]
```diff
@@ -54,9 +54,9 @@ attribute is used.
 user for this identity. First non-empty attribute is used.
 <7> Optional DN to use to bind during the search phase. Must be set if
 `bindPassword` is defined.
-<8> Optional reference to an {product-title} Secret containing the bind
+<8> Optional reference to an {product-title} `Secret` containing the bind
 password. Must be set if `bindDN` is defined.
-<9> Optional: Reference to an {product-title} ConfigMap containing the
+<9> Optional: Reference to an {product-title} `ConfigMap` containing the
```
Comment (jboxman) [inline modules/identity-provider-ldap-CR.adoc:59]: object

File: modules/identity-provider-ldap-CR.adoc [hunk 34]
```diff
@@ -54,9 +54,9 @@ attribute is used.
 user for this identity. First non-empty attribute is used.
 <7> Optional DN to use to bind during the search phase. Must be set if
 `bindPassword` is defined.
-<8> Optional reference to an {product-title} Secret containing the bind
+<8> Optional reference to an {product-title} `Secret` containing the bind
```
Comment (jboxman) [inline modules/identity-provider-ldap-CR.adoc:57]: object

**Review by jboxman — APPROVED** [review]
Grab bag of mostly minor observations.

File: authentication/identity_providers/configuring-basic-authentication-identity-provider.adoc [hunk 35]
```diff
@@ -1,12 +1,12 @@
...
-= Configuring an basic authentication identity provider
+= Configuring a basic authentication identity provider
 include::modules/common-attributes.adoc[]
 :context: configuring-basic-authentication-identity-provider
 toc::[]
 
 Configure a `basic-authentication` identity provider for users to log in to
 {product-title} with credentials validated against a remote identity provider.
-Basic authentication is a generic backend integration mechanism.
+Basic authentication is a generic back-end integration mechanism.
```
[the remark this answers is not in the document]
Comment (bergerhoffer) [reply authentication/identity_providers/configuring-basic-authentication-identity-provider.adoc:9]: When back-end is an adjective, it should be hyphenated according to IBM style guide:

>  back-end adj
Write as shown - hyphenated. If possible use a more specific term, such as “server,” “operating system,” or “network.”

When it's a noun (as in your example), it's two words:

> back end n
Write as shown - two words. If possible use a more specific term, such as “server,” “operating system,” or “network.”

File: modules/security-context-constraints-rbac.adoc (hunk 22 above)
Comment (bergerhoffer) [reply modules/security-context-constraints-rbac.adoc:63]: Not sure, so not changing with the scope of this PR being for formatting updates.

**Pull request merged**